ui/app/AppLayouts/Chat/components/MessageContextMenu.qml

```diff
@@ -0,0 +1,124 @@
+import QtQuick 2.12
+import QtQuick.Controls 2.3
+import QtQuick.Layouts 1.3
+import QtQml.Models 2.3
+import "../../../../imports"
+import "../../../../shared"
+import "./"
+
+PopupMenu {
+    property bool isProfile: false
+
+    id: messageContextMenu
+    width: messageContextMenu.isProfile ? profileHeader.width : emojiContainer.width
+
+    Item {
+        id: emojiContainer
+        visible: !messageContextMenu.isProfile
+        width: emojiRow.width
+        height: visible ? emojiRow.height : 0
+
+        Row {
+            id: emojiRow
+            spacing: Style.current.smallPadding
+            leftPadding: Style.current.smallPadding
+            rightPadding: Style.current.smallPadding
+            bottomPadding: Style.current.padding
+
+            Repeater {
+                model: reactionModel
+                delegate: EmojiReaction {
+                    source: "../../../img/" + filename
+                    emojiId: model.emojiId
+                }
+            }
+        }
+    }
+
+    Rectangle {
+        property bool hovered: false
+
+        id: profileHeader
+        visible: messageContextMenu.isProfile
+        width: 200
+        height: visible ? profileImage.height + username.height + Style.current.padding : 0
+        color: hovered ? Style.current.secondaryBackground : Style.current.transparent
+
+        Rectangle {
+            id: profileImage
+            width: 40
+            height: 40
+            border.width: 1
+            border.color: Style.current.border
+            radius: 50
+            anchors.top: parent.top
+            anchors.topMargin: 4
+            anchors.horizontalCenter: parent.horizontalCenter
+
+            Image {
+                id: identiconImage
+                width: 40
+                height: 40
+                fillMode: Image.PreserveAspectFit
+                source: profilePopup.identicon
+                mipmap: true
+                smooth: false
+                antialiasing: true
+            }
+        }
```

Comment:
Just an idea: can't we use our `Identicon` or `RoundIcon` components for this?

Reply:
Done